infrastructure/api_management.tf

```diff
@@ -26,3 +27,64 @@ __________________________________*/
   tags = var.tags
 
 }
+
+/*________________________________
+| API Management Public IP Address |
+__________________________________*/
+
+module "apim-public-ip" {
+  for_each = length(var.apim_config.zones) > 0 ? { for key, value in var.regions : key => value } : {}
+
+  source = "../../dtos-devops-templates/infrastructure/modules/public-ip"
+
+  name                = "${module.config[each.key].names.public-ip-address}-api-mgmt"
+  resource_group_name = azurerm_resource_group.rg_hub[each.key].name
+  location            = each.key
+  allocation_method   = var.apim_config.public_ip_allocation_method
+  domain_name_label   = module.config[each.key].names.api-management
+  sku                 = var.apim_config.public_ip_sku
+  zones               = var.apim_config.zones
+
+  tags = var.tags
+}
+
+
+/*______________________________________
+| API Management Private DNS A Records |
+______________________________________*/
+
+module "apim-private-dns-a-records" {
+  for_each = local.custom_domains_map
+
+
+  source = "../../dtos-devops-templates/infrastructure/modules/private-dns-a-record"
+
+  name                = each.value.name
+  resource_group_name = "${module.config[each.value.region].names.resource-group}-private-dns-zones"
+  private_dns_a_record = {
+    zone_name    = module.private_dns_zone_private_nationalscreening_nhs_uk[each.value.region].name
+    a_record_ttl = each.value.ttl
+    ip_address   = module.api-management[each.value.region].private_ip_address
+  }
+
+  tags = var.tags
+}
+
+
+locals {
+
+  custom_domains = flatten([
+    for region_key, region_val in var.regions : [
```

Comment:
`for region_key in keys(var.regions) :`.   

or 

`for region_key, _ in var.regions`